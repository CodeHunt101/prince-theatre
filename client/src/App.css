.App {
  text-align: center;
}

/* ::::Header:::: */
.highlight-cheapest {
  color: white;
  background-color: #4caf50;
}

.highlight-expensive {
  color: #ff1744;
}
