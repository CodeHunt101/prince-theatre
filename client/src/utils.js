export const capitalise = (string) => (
  string[0].toUpperCase() + string.slice(1).toLowerCase()
)

export const handleErrors = (response) => {
  if (!response.ok) {
      throw Error(response.statusText);
  }
  return response;
}