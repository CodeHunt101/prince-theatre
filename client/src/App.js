import { Header } from './components/Header';
import { Movies } from './components/Movies';
import './App.css';
import { Container, Box, Paper } from '@mui/material';

function App() {
  
  return (
    <div className="App">
      <Container maxWidth="lg">
        <Box component={Paper}>
          <Box sx={{pt:5, pb:5}} >
            <Header />
            <Movies />
          </Box>
        </Box>
      </Container>
    </div>
  );
}

export default App;
