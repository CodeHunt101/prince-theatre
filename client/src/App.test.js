import App from "./App"
import { capitalise } from "./utils"
import renderer from "react-test-renderer"

describe("Helpers", () => {
  test("capitalises the movies endpoints", () => {
    expect(capitalise("cinema")).toBe("Cinema")
    expect(capitalise("CINEMA")).toBe("Cinema")
    expect(capitalise("cINEMA")).toBe("Cinema")
    expect(capitalise("cInEmA")).toBe("Cinema")
  })
})

test("App renders correctly", () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
