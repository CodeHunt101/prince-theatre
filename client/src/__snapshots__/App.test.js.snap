// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiBox-root css-gx1m5n-MuiPaper-root"
    >
      <div
        className="MuiBox-root css-1kgakwq"
      >
        <header
          className="MuiBox-root css-0"
        >
          <h1
            className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
          >
            <b>
              Prince's Theatre
            </b>
          </h1>
          <h4
            className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
          >
            Save your money, guaranteed!
          </h4>
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1idk476-MuiTypography-root"
          >
            Hi there! Welcome to Prince's Theatre, here you'll see some movie prices from two cinemas!
          </p>
        </header>
        <section
          circle="true"
          className="MuiGrid-root MuiGrid-item css-48pxfh-MuiGrid-root"
          id="movie-cards-list"
        >
          <div
            className="MuiContainer-root MuiContainer-maxWidthLg css-1t9p3vr-MuiContainer-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-qktg34-MuiGrid-root"
            >
              <div
                className="MuiBox-root css-1dwuyr9"
              >
                <span
                  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
                  role="progressbar"
                >
                  <span
                    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
                    style={Object {}}
                  />
                  <span
                    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
                    style={Object {}}
                  />
                </span>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
